quesma/schema/registry.go

```diff
@@ -145,13 +185,17 @@ func (s *schemaRegistry) populateAliases(indexConfiguration config.IndexConfigur
 	}
 }
 
-func (s *schemaRegistry) populateSchemaFromTableDefinition(definitions map[string]Table, indexName string, fields map[FieldName]Field) (existsInDataSource bool) {
+func (s *schemaRegistry) populateSchemaFromTableDefinition(definitions map[string]Table, indexName string, fields map[FieldName]Field, internalToPublicFieldsEncodings map[string]string) (existsInDataSource bool) {
 	tableDefinition, found := definitions[indexName]
 	if found {
 		logger.Debug().Msgf("loading schema for table %s", indexName)
 
 		for _, column := range tableDefinition.Columns {
-			propertyName := FieldName(strings.Replace(column.Name, "::", ".", -1))
+			propertyName := FieldName(internalToPublicFieldsEncodings[column.Name])
```

Comment:
ugh, that's really a publicFieldName, not a property, but it blends with that Field structur defined in `schema.go` which overall collides a little with the map of encodings

Reply:
yes, we can try to fix that later